From 5d95b77632acbd4dd29dab898140b731766235b3 Mon Sep 17 00:00:00 2001
From: Asif Ikbal <asif.ikbal@amd.com>
Date: Mon, 13 Jul 2026 14:24:09 +0530
Subject: [LINUX PATCH] v4l: xilinx: vpss-scaler: Add YUV422 and YUV420 12 bit
 media bus formats

Add MEDIA_BUS_FMT_UYVY12_1X24 (YUV422) and MEDIA_BUS_FMT_UYYVYY12_4X24
(YUV420) to the scaler input and output format setup.

Signed-off-by: Asif Ikbal <asif.ikbal@amd.com>
---
 drivers/media/platform/xilinx/xilinx-vpss-scaler.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
index a0ca0625c7a8..4b81bda572cc 100644
--- a/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
+++ b/drivers/media/platform/xilinx/xilinx-vpss-scaler.c
@@ -1398,12 +1398,14 @@ xv_vscaler_setup_video_fmt(struct xscaler_device *xscaler, u32 code_in)
 	switch (code_in) {
 	case MEDIA_BUS_FMT_VYYUYY8_1X24:
 	case MEDIA_BUS_FMT_VYYUYY10_4X20:
+	case MEDIA_BUS_FMT_UYYVYY12_4X24:
 		dev_dbg(xscaler->xvip.dev,
 			"Vscaler Input Media Format YUV 420");
 		video_in = XVIDC_CSF_YCRCB_420;
 		break;
 	case MEDIA_BUS_FMT_UYVY8_1X16:
 	case MEDIA_BUS_FMT_UYVY10_1X20:
+	case MEDIA_BUS_FMT_UYVY12_1X24:
 		dev_dbg(xscaler->xvip.dev,
 			"Vscaler Input Media Format YUV 422");
 		video_in = XVIDC_CSF_YCRCB_422;
@@ -1470,12 +1472,14 @@ static int xv_hscaler_setup_video_fmt(struct xscaler_device *xscaler,
 	switch (code_out) {
 	case MEDIA_BUS_FMT_VYYUYY8_1X24:
 	case MEDIA_BUS_FMT_VYYUYY10_4X20:
+	case MEDIA_BUS_FMT_UYYVYY12_4X24:
 		dev_dbg(xscaler->xvip.dev,
 			"Hscaler Output Media Format YUV 420\n");
 		video_out = XVIDC_CSF_YCRCB_420;
 		break;
 	case MEDIA_BUS_FMT_UYVY8_1X16:
 	case MEDIA_BUS_FMT_UYVY10_1X20:
+	case MEDIA_BUS_FMT_UYVY12_1X24:
 		dev_dbg(xscaler->xvip.dev,
 			"Hscaler Output Media Format YUV 422\n");
 		video_out = XVIDC_CSF_YCRCB_422;
-- 
2.34.1

